MetadataExtractor/IO/BufferReader.cs

```diff
@@ -65,14 +69,13 @@ public bool TryAdvance(int n)
 
     public sbyte GetSByte()
     {
-        return unchecked((sbyte)GetByte());
+        return unchecked((sbyte)_bytes[_position++]);
     }
 
     public ushort GetUInt16()
     {
-        Span<byte> bytes = stackalloc byte[2];
-
-        GetBytes(bytes);
+        var bytes = _bytes.Slice(_position, 2);
+        _position += 2;
```

Comment:
This means we're no longer validating length on several `Get` methods, like this and those that follow.

Could we change `Advance` to validate, return the span, and progress the reader? With that, we'd use `Advance(2)` here, and so on below.

Reply:
updated.